validation/genesis/genesis-predeploy_test.go

```diff
@@ -0,0 +1,230 @@
+package genesis
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/stretchr/testify/require"
+)
+
+// TODO deduplicate this
+// perChainTestName ensures test can easily be filtered by chain name or chain id using the -run=regex testflag.
+func perChainTestName(chain *superchain.ChainConfig) string {
+	return chain.Name + fmt.Sprintf(" (%d)", chain.ChainID)
+}
+
+func TestGenesisPredeploys(t *testing.T) {
+	for _, chain := range OPChains {
+		if chain.SuperchainLevel == Standard || chain.StandardChainCandidate {
+			t.Run(perChainTestName(chain), func(t *testing.T) {
+				// Do not run in parallel
+				testGenesisPredeploys(t, chain)
+			})
+		}
+	}
+}
+
+// Invoke this with go test -timeout 0 ./validation/genesis -run=TestGenesisPredeploys -v
+// REQUIREMENTS:
+// pnpm and yarn, so we can prepare https://codeload.github.com/Saw-mon-and-Natalie/clones-with-immutable-args/tar.gz/105efee1b9127ed7f6fedf139e1fc796ce8791f2
+func testGenesisPredeploys(t *testing.T, chain *ChainConfig) {
+	chainId := chain.ChainID
+
+	vis, ok := ValidationInputs[chainId]
+
+	if !ok {
+		t.Skip("WARNING: cannot yet validate this chain (no validation metadata)")
+	}
```

Comment:
TODO we should update this before merging the PR (the test should just fail if the metadata is not present).